lib/types/described_type.js

```diff
@@ -29,11 +27,15 @@ var debug = require('debug')('amqp10:types:described');
  * (Note: this example shows a string-typed descriptor, which should be considered reserved)
  *
  * @constructor
- * @param descriptor        Descriptor for the type (can be any valid AMQP type, including another described type).
+ * @param descriptorOrType  Descriptor for the type (can be any valid AMQP type, including another described type), or the type itself.
  * @param value             Value of the described type (can also be any valid AMQP type, including another described type).
  */
-function DescribedType(descriptor, value) {
-  this.descriptor = descriptor;
+function DescribedType(descriptorOrType, value) {
+  if (descriptorOrType && descriptorOrType.prototype && descriptorOrType.prototype.Descriptor)
```

Comment:
Javascript needs C#'s Elvis operator, then this'd be:
`this.descriptor = descriptorOrType?.prototype?.Descriptor?.code ?? descriptorOrType;`